packages/nimble-components/src/banner/styles.ts

```diff
@@ -0,0 +1,214 @@
+import { css } from '@microsoft/fast-element';
+import { display } from '@microsoft/fast-foundation';
+import {
+    BannerFail100DarkUi,
+    Black75,
+    Fail100LightUi,
+    Information100DarkUi,
+    Information100LightUi,
+    Warning100DarkUi,
+    Warning100LightUi,
+    White
+} from '@ni/nimble-tokens/dist/styledictionary/js/tokens';
+
+import {
+    actionRgbPartialColor,
+    applicationBackgroundColor,
+    bodyFont,
+    borderHoverColor,
+    buttonLabelFontColor,
+    controlHeight,
+    controlSlimHeight,
+    fillSelectedColor,
+    iconColor,
+    iconSize,
+    linkActiveFontColor,
+    linkDisabledFontColor,
+    linkFontColor,
+    standardPadding
+} from '../theme-provider/design-tokens';
+import { Theme } from '../theme-provider/types';
+import { hexToRgbaCssColor } from '../utilities/style/colors';
+import { themeBehavior } from '../utilities/style/theme';
+
+export const styles = css`
+    ${display('flex')}
+
+    :host {
+        width: 100%;
+        font: ${bodyFont};
+        font-size: 12.8px;
+        align-items: top;
+        overflow: hidden;
+        color: ${White};
+        ${iconColor.cssCustomProperty}: ${hexToRgbaCssColor(White, 0.6)};
+    }
+
+    :host(:not([open])) {
+        display: none;
+    }
+
+    .container {
+        display: flex;
+        width: 100%;
+    }
+
+    .icon {
+        width: 48px;
+        display: flex;
+        justify-content: center;
+        margin-top: 8px;
+        flex: 0 0 auto;
+    }
+
+    .text {
+        display: inline;
+        margin-top: 7px;
+        margin-bottom: 7px;
+    }
+
+    slot[name='title']::slotted(*) {
+        font-weight: bold;
+        padding-right: 8px;
+        white-space: nowrap;
+    }
+
+    :host([title-hidden]) slot[name='title'] {
+        ${
+            /**
+             * Hide content visually while keeping it screen reader-accessible.
+             * Source: https://webaim.org/techniques/css/invisiblecontent/#techniques
+             * See discussion here: https://github.com/microsoft/fast/issues/5740#issuecomment-1068195035
+             */
+            ''
+        }
+        display: inline-block;
+        height: 1px;
+        width: 1px;
+        position: absolute;
+        margin: -1px;
+        clip: rect(1px, 1px, 1px, 1px);
+        clip-path: inset(50%);
+        overflow: hidden;
+        padding: 0;
+    }
```

Comment:
I don't think this comment actually makes sense here. We aren't keeping the title visually hidden to make it screen reader accessible, i.e. accessible on the accessibility tree. In fact that seems like an actual problem, we see the banner title text and this text in the accessibility tree:
![image](https://user-images.githubusercontent.com/1588923/218884084-bd837be0-27f1-4e0b-9b45-8e3b8b5f7305.png)

It was being visually hidden to make it detected by innerText. If we were keeping it in this state then we would need to set `aria-hidden="true"` on it because we don't want it duplicated in the accessibility tree.

Anyway shouldn't be needed after switching to textContent. This can just be set to display none. That will remove it from the accessibility tree as well when it is hidden. May still need to set it to aria-hidden so it doesn't show up twice in the tree when visible.

Reply:
Thanks, good points.

Reply:
After switching to using `aria-labelledby` (to get the accessible name to update when the title is updated), this original approach is actually necessary to keep the accessible name even when the title is hidden.